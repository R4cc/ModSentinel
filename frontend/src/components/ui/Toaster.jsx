import { Toaster as SonnerToaster } from "sonner";

export function Toaster() {
  return (
    <SonnerToaster
      position="top-center"
      richColors
      closeButton
      containerClassName="fixed top-0 z-50 flex w-full justify-center"
      toastOptions={{
        className:
          "w-full max-w-[420px] rounded-md border bg-background p-4 text-sm text-foreground shadow-md",
        descriptionClassName: "text-muted-foreground",
        actionButtonClassName:
          "bg-primary text-primary-foreground hover:bg-primary/90",
        cancelButtonClassName:
          "bg-muted text-muted-foreground hover:bg-muted/90",
      }}
    />
  );
}

export default Toaster;
