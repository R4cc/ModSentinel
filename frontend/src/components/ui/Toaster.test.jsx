import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/vitest";
import { describe, it, expect, vi } from "vitest";
import Toaster from "./Toaster.jsx";

const { toast } = await vi.importActual("@/lib/toast.ts");

describe("Toaster", () => {
  it("renders container once", async () => {
    render(<Toaster />);
    toast.info("hi");
    await waitFor(() =>
      expect(
        document.querySelectorAll('section[aria-label="Notifications alt+T"]'),
      ).toHaveLength(1),
    );
    const el = document.querySelector(
      'section[aria-label="Notifications alt+T"]',
    );
    expect(el).toBeInTheDocument();
  });
});
