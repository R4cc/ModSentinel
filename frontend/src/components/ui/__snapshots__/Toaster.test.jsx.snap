// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toaster > renders container once 1`] = `
<div>
  <section
    aria-atomic="false"
    aria-label="Notifications alt+T"
    aria-live="polite"
    aria-relevant="additions text"
    tabindex="-1"
  >
    <ol
      data-sonner-theme="light"
      data-sonner-toaster="true"
      data-x-position="center"
      data-y-position="top"
      dir=""
      style="--front-toast-height: 0px; --width: 356px; --gap: 14px; --offset-top: 24px; --offset-right: 24px; --offset-bottom: 24px; --offset-left: 24px; --mobile-offset-top: 16px; --mobile-offset-right: 16px; --mobile-offset-bottom: 16px; --mobile-offset-left: 16px;"
      tabindex="-1"
    >
      <li
        class="w-full max-w-[420px] rounded-md border bg-background p-4 text-sm text-foreground shadow-md"
        data-dismissible="true"
        data-expanded="false"
        data-front="true"
        data-index="0"
        data-mounted="true"
        data-promise="false"
        data-removed="false"
        data-rich-colors="true"
        data-sonner-toast=""
        data-styled="true"
        data-swipe-out="false"
        data-swiped="false"
        data-swiping="false"
        data-visible="true"
        data-x-position="center"
        data-y-position="top"
        style="--index: 0; --toasts-before: 0; --z-index: 1; --offset: 0px; --initial-height: 0px;"
        tabindex="0"
      >
        <button
          aria-label="Close toast"
          class=""
          data-close-button="true"
          data-disabled="false"
        >
          <svg
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="6"
              y1="6"
              y2="18"
            />
            <line
              x1="6"
              x2="18"
              y1="6"
              y2="18"
            />
          </svg>
        </button>
        <div
          class=""
          data-content=""
        >
          <div
            class=""
            data-title=""
          >
            hi
          </div>
        </div>
      </li>
    </ol>
  </section>
</div>
`;
